recipes/openmathreasoning/scripts/merge_new_summary.py

```diff
@@ -94,8 +96,8 @@ def trim_reasoning_generation(reasoning_generation, start_tag, end_tag, strict_e
                 reasoning_trace[rightmost_match.end():].split("\n\n")[0]
             )
 
-            # If the end tag is not present, add it
-            if end_tag not in reasoning_trace:
+            # If we originally had tags and the end tag is not present, add it
+            if start_tag_position != -1 and end_tag not in reasoning_trace:
```

Comment:
We will most likely not reach this code if we structure the above `if else` part.